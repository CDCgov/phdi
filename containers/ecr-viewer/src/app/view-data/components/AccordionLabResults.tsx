import { Accordion, HeadingLevel, Tag } from "@trussworks/react-uswds";
import React from "react";

interface AccordionLabResultsProps {
  title: string;
  abnormalTag: boolean;
  content: React.JSX.Element[];
  organizationId: string;
  collapsedByDefault?: boolean;
  headingLevel?: HeadingLevel;
}

/**
 * Accordion component for displaying lab results.
 * @param props - The props object.
 * @param props.title - The title of the lab result.
 * @param props.abnormalTag - Boolean value if the lab result is abnormal.
 * @param props.content - The content within the accordian.
 * @param props.organizationId - The id of the organization you are getting lab results for.
 * @param props.collapsedByDefault - Whether or not to collapse by default for the accordion
 * @param props.headingLevel - Heading level for the Accordion menu title.
 * @returns React element representing the AccordionLabResults component.
 */
export const AccordionLabResults: React.FC<AccordionLabResultsProps> = ({
  title,
  abnormalTag,
  content,
  organizationId,
  collapsedByDefault = false,
  headingLevel = "h5"
}: AccordionLabResultsProps): React.JSX.Element => {
  return (
    <Accordion
      items={[
        {
          title: (
            <>
              {title}
              {abnormalTag && (
                <Tag background={"#B50909"} className={"margin-left-105"}>
                  Abnormal
                </Tag>
              )}
            </>
          ),
          content: content,
          expanded: collapsedByDefault,
          id: title,
          headingLevel,
          className: `acc_item_${organizationId}`,
        },
      ]}
      className={`accordion-rr accordion_${organizationId} margin-bottom-3`}
    />
  );
};
