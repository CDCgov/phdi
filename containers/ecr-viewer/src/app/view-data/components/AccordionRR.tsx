import { Accordion, Icon, Tag } from "@trussworks/react-uswds";
import React from "react";

interface AccordionRRProps {
  title: string;
  abnormalTag: boolean;
  content: React.JSX.Element;
}
export const AccordionRR: React.FC<AccordionRRProps> = ({
  title,
  abnormalTag,
  content,
}) => {
  return (
    <Accordion
      items={[
        {
          title: (
            <>
              {title}
              {abnormalTag && (
                <Tag background={"#B50909"} className={"margin-left-105"}>
                  Abnormal
                </Tag>
              )}
              <Icon.Remove className={"minimize-container"} size={3} />
            </>
          ),
          content: content,
          expanded: true,
          id: "123",
          headingLevel: "h4",
        },
      ]}
      className={"accordion-rr"}
    />
  );
};
