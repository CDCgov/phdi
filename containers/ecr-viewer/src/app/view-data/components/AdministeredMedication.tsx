import { noData } from "@/app/utils";
import { Table } from "@trussworks/react-uswds";
import { formatDate } from "@/app/services/formatService";

type AdministeredMedicationProps = {
  medicationData: AdministeredMedicationData[];
};
export type AdministeredMedicationData = {
  name: string;
  date?: string;
};

/**
 * Returns a table displaying administered medication information.
 * @param props - Props for the component.
 * @param props.medicationData - Array of data of medicine name and start date
 * @returns The JSX element representing the table, or undefined if no administered medications are found.
 */
export const AdministeredMedication = ({
  medicationData,
}: AdministeredMedicationProps) => {
  if (!medicationData?.length) {
    return null;
  }

  const header = ["Medication Name", "Medication Start Date"];
  return (
    <Table
      bordered={false}
      fullWidth={true}
      caption="Administered Medications"
      className={
        "table-caption-margin margin-y-0 border-top border-left border-right"
      }
      data-testid="table"
    >
      <thead>
        <tr>
          {header.map((column) => (
            <th key={`${column}`} scope="col" className="bg-gray-5 minw-15">
              {column}
            </th>
          ))}
        </tr>
      </thead>
      <tbody>
        {medicationData.map((entry, index: number) => {
          return (
            <tr key={`table-row-${index}`}>
              <td>{entry?.name ?? noData}</td>
              <td>{formatDate(entry?.date) ?? noData}</td>
            </tr>
          );
        })}
      </tbody>
    </Table>
  );
};
