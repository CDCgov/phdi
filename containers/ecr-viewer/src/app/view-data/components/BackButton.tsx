import React from "react";
import { Icon } from "@trussworks/react-uswds";
import Link from "next/link";

/**
 * Back button component for returning users from the eCR Viewer page to the eCR Library home page
 * @returns <BackButton/> a react component back button
 */
export const BackButton = () => {
  const isNonIntegratedViewer =
    process.env.NEXT_PUBLIC_NON_INTEGRATED_VIEWER === "true";

  return (
    <div className="display-flex back-button-wrapper">
      {isNonIntegratedViewer ? (
        <Link href={"/"} className={"back-button display-flex"}>
          <Icon.ArrowBack aria-label={"Back Arrow"} size={3} />
          Back to eCR Library
        </Link>
      ) : (
        ""
      )}
    </div>
  );
};
