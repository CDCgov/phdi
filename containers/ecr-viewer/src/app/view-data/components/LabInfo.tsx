import { DataDisplay, DisplayDataProps } from "@/app/utils";
import {
  AccordianSection,
  AccordianH4,
  AccordianDiv,
} from "../component-utils";
import React from "react";
import { LabReportElementData } from "@/app/labs/utils";
import { ExpandCollapseButtons } from "@/app/view-data/components/ExpandCollapseButtons";
import { truncateLabNameWholeWord, formatString } from "@/app/formatService";

interface LabInfoProps {
  labResults: LabReportElementData[];
}

/**
 * Renders lab information and RR info in an accordion section.
 * @param props - The props object.
 * @param props.labResults - Array of Lab result items.
 * @returns React element representing the LabInfo component.
 */
export const LabInfo = ({ labResults }: LabInfoProps): React.JSX.Element => {
  const renderLabInfo = () => {
    return (
      <>
        {labResults.map((labResult, labIndex) => {
          // This is to build the selector based off if orgId exists
          // Sometimes it doesn't, so we default to the base class
          // the orgId makes it so that when you have multiple, it can distinguish
          // which org it is modifying
          const accordionSelectorClass = labResult.organizationId
            ? `.accordion_${labResult.organizationId}`
            : ".accordion-rr";
          const buttonSelectorClass = labResult.organizationId
            ? `.acc_item_${labResult.organizationId}`
            : "h5";
          const labName = `Lab Results from 
                ${truncateLabNameWholeWord(
                  labResult.organizationDisplayDataProps[0].value as string,
                )}`;
          return (
            <div key={`${labResult.organizationId}${labIndex}`}>
              <AccordianH4 id={formatString(labName)}>{labName}</AccordianH4>
              <AccordianDiv>
                {labResult?.organizationDisplayDataProps?.map(
                  (item: DisplayDataProps, index: any) => {
                    if (item.value)
                      return <DataDisplay item={item} key={index} />;
                  },
                )}
                <div className="display-flex">
                  <div className={"margin-left-auto padding-top-1"}>
                    <ExpandCollapseButtons
                      id={"lab-info"}
                      buttonSelector={`${buttonSelectorClass} > .usa-accordion__button`}
                      accordionSelector={`${accordionSelectorClass} > .usa-accordion__content`}
                      expandButtonText={"Expand all labs"}
                      collapseButtonText={"Collapse all labs"}
                    />
                  </div>
                </div>
                {labResult.diagnosticReportDataElements}
              </AccordianDiv>
            </div>
          );
        })}
      </>
    );
  };

  return (
    <AccordianSection>
      {labResults.length > 0 && renderLabInfo()}
    </AccordianSection>
  );
};

export default LabInfo;
