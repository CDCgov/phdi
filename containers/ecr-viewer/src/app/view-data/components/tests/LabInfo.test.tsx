import { render, screen } from "@testing-library/react";
import LabInfo from "@/app/view-data/components/LabInfo";
import userEvent from "@testing-library/user-event";
import { AccordionLabResults } from "@/app/view-data/components/AccordionLabResults";
import React from "react";
import { DisplayData } from "@/app/utils";

describe("LabInfo", () => {
  const labinfoOrg: DisplayData[] = [
    {
      title: "Lab Name",
      value: "Org1",
    },
    {
      title: "Lab Name",
      value: "Org2",
    },
  ];
  const labInfoJsx = () => (
    <LabInfo
      labResults={[
        {
          diagnosticReportDataElements: [
            <AccordionLabResults
              key={"blah"}
              title={"ph of urine strip"}
              abnormalTag={false}
              content={[<div key={"1"}>5</div>]}
            />,
          ],
          organizationDisplayData: [labinfoOrg[0]],
        },
        {
          diagnosticReportDataElements: [
            <AccordionLabResults
              key={"blah2"}
              title={"ph of saliva"}
              abnormalTag={false}
              content={[<div key={"2"}>7</div>]}
            />,
          ],
          organizationDisplayData: [labinfoOrg[1]],
        },
      ]}
    />
  );
  it("should hide all labs when collapse button is clicked", async () => {
    const user = userEvent.setup();

    render(labInfoJsx());
    const collapseButtons = screen.getAllByText("Collapse all labs");
    collapseButtons.forEach(async (button) => {
      console.log(button);
      await user.click(button);
    });
    screen.debug();
    screen
      .getAllByTestId("accordionButton", { exact: false })
      .forEach((button) => {
        expect(button).toHaveAttribute("aria-expanded", "false");
      });
    screen
      .getAllByTestId("accordionItem", { exact: false })
      .forEach((accordion) => {
        expect(accordion).toHaveAttribute("hidden", "true");
      });
  });
  it("should hide all labs when collapse button is clicked", async () => {
    const user = userEvent.setup();

    render(labInfoJsx());
    const collapseButtons = screen.getAllByText("Collapse all labs");
    collapseButtons.forEach(async (button) => {
      await user.click(button);
    });
    screen
      .getAllByTestId("accordionButton", { exact: false })
      .forEach((button) => {
        expect(button).toHaveAttribute("aria-expanded", "false");
      });
    screen
      .getAllByTestId("accordionItem", { exact: false })
      .forEach((accordion) => {
        expect(accordion).toHaveAttribute("hidden", "true");
      });

    await user.click(screen.getByText("Expand all labs"));
    screen
      .getAllByTestId("accordionButton", { exact: false })
      .forEach((button) => {
        expect(button).toHaveAttribute("aria-expanded", "true");
      });
    screen
      .getAllByTestId("accordionItem", { exact: false })
      .forEach((accordion) => {
        expect(accordion).not.toHaveAttribute("hidden");
      });
  });
});
