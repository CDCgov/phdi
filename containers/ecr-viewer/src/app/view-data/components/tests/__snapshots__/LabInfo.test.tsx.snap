// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabInfo should match snapshot test 1`] = `
<div>
  <div
    class="margin-top-0"
  >
    <div
      class="padding-bottom-3"
    >
      <div
        class="usa-summary-box__body"
      >
        <div>
          <h4
            class="usa-summary-box__heading padding-y-105"
            id="lab-results-from-providence-st-joseph-medical"
          >
            Lab Results from 
                PROVIDENCE ST. JOSEPH MEDICAL
          </h4>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title"
                >
                  Lab Performing Name
                </div>
                <div
                  class="grid-col-auto maxw7 text-pre-line"
                >
                  PROVIDENCE ST. JOSEPH MEDICAL CENTER LABORATORY (CLIA 05D0672675)
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title"
                >
                  Lab Address
                </div>
                <div
                  class="grid-col-auto maxw7 text-pre-line"
                >
                  501 S. Buena Vista Street
Burbank, CA
91505
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title"
                >
                  Number of Results
                </div>
                <div
                  class="grid-col-auto maxw7 text-pre-line"
                >
                  2
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div
              class="display-flex"
            >
              <div
                class="margin-left-auto padding-top-1"
              >
                <button
                  class="usa-button usa-button--unstyled"
                  data-testid="button"
                  id="lab-info-expand-button"
                  type="button"
                >
                  Expand all labs
                </button>
                <span
                  class="vertical-line"
                />
                <button
                  class="usa-button usa-button--unstyled"
                  data-testid="button"
                  id="lab-info-collapse-button"
                  type="button"
                >
                  Collapse all labs
                </button>
              </div>
            </div>
            <div
              class="usa-accordion accordion-rr accordion_9e215f4e-aac1-10cb-e412-020cd13a6ad9 margin-bottom-3"
              data-testid="accordion"
            >
              <h5
                class="usa-accordion__heading acc_item_9e215f4e-aac1-10cb-e412-020cd13a6ad9"
              >
                <button
                  aria-controls="STOOL PATHOGENS, NAAT, 12 TO 25 TARGETS"
                  aria-expanded="false"
                  class="usa-accordion__button"
                  data-testid="accordionButton_STOOL PATHOGENS, NAAT, 12 TO 25 TARGETS"
                  type="button"
                >
                  STOOL PATHOGENS, NAAT, 12 TO 25 TARGETS
                  <span
                    class="usa-tag margin-left-105"
                    data-testid="tag"
                    style="background: rgb(181, 9, 9);"
                  >
                    Abnormal
                  </span>
                </button>
              </h5>
              <div
                class="usa-accordion__content usa-prose acc_item_9e215f4e-aac1-10cb-e412-020cd13a6ad9"
                data-testid="accordionItem_STOOL PATHOGENS, NAAT, 12 TO 25 TARGETS"
                hidden=""
                id="STOOL PATHOGENS, NAAT, 12 TO 25 TARGETS"
              >
                <table
                  class="usa-table usa-table--borderless width-full src-components-Table-Table-module__fixed--Eq00W table-caption-margin margin-y-0"
                  data-testid="table"
                >
                  <thead>
                    <tr>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Component
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Value
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Ref Range
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Test Method
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Lab Comment
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td
                        class="text-top"
                      >
                        Campylobacter, NAAT
                      </td>
                      <td
                        class="text-top"
                      >
                        Not Detected
                      </td>
                      <td
                        class="text-top"
                      >
                        Not Detected
                      </td>
                      <td
                        class="text-top"
                      >
                        LAB DEVICE: BIOFIRE速 FILMARRAY速 2.0 SYSTEM
                      </td>
                      <td
                        class="text-top"
                      >
                        <span
                          class="text-italic text-base"
                        >
                          No data
                        </span>
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="text-top"
                      >
                        Plesiomonas shigelloides, NAAT
                      </td>
                      <td
                        class="text-top"
                      >
                        Not Detected
                      </td>
                      <td
                        class="text-top"
                      >
                        Not Detected
                      </td>
                      <td
                        class="text-top"
                      >
                        LAB DEVICE: BIOFIRE速 FILMARRAY速 2.0 SYSTEM
                      </td>
                      <td
                        class="text-top"
                      >
                        <span
                          class="text-italic text-base"
                        >
                          No data
                        </span>
                      </td>
                    </tr>
                  </tbody>
                </table>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Analysis Time
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      09/28/2022 2:00 PM PDT
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Collection Time
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      09/28/2022 8:51 PM UTC
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Received Time
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      09/28/2022 8:51 PM UTC
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Specimen (Source)
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      Stool
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Anatomical Location/Laterality
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      STOOL SPECIMEN / Unknown
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Collection Method/Volume
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      <span
                        class="no-data text-italic text-base"
                      >
                        No data
                      </span>
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Resulting Agency Comment
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      <span
                        class="no-data text-italic text-base"
                      >
                        No data
                      </span>
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Authorizing Provider
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      Ambhp1 Test MD
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Result Type
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      MICROBIOLOGY - GENERAL ORDERABLES
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Narrative
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      <span
                        class="no-data text-italic text-base"
                      >
                        No data
                      </span>
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
              </div>
            </div>
            <div
              class="usa-accordion accordion-rr accordion_9e215f4e-aac1-10cb-e412-020cd13a6ad9 margin-bottom-3"
              data-testid="accordion"
            >
              <h5
                class="usa-accordion__heading acc_item_9e215f4e-aac1-10cb-e412-020cd13a6ad9"
              >
                <button
                  aria-controls="STOOL PATHOGENS, NAAT, 12 TO 25 TARGETS"
                  aria-expanded="false"
                  class="usa-accordion__button"
                  data-testid="accordionButton_STOOL PATHOGENS, NAAT, 12 TO 25 TARGETS"
                  type="button"
                >
                  STOOL PATHOGENS, NAAT, 12 TO 25 TARGETS
                </button>
              </h5>
              <div
                class="usa-accordion__content usa-prose acc_item_9e215f4e-aac1-10cb-e412-020cd13a6ad9"
                data-testid="accordionItem_STOOL PATHOGENS, NAAT, 12 TO 25 TARGETS"
                hidden=""
                id="STOOL PATHOGENS, NAAT, 12 TO 25 TARGETS"
              >
                <table
                  class="usa-table usa-table--borderless width-full src-components-Table-Table-module__fixed--Eq00W table-caption-margin margin-y-0"
                  data-testid="table"
                >
                  <thead>
                    <tr>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Component
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Value
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Ref Range
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Test Method
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Lab Comment
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td
                        class="text-top"
                      >
                        Campylobacter, NAAT
                      </td>
                      <td
                        class="text-top"
                      >
                        Not Detected
                      </td>
                      <td
                        class="text-top"
                      >
                        Not Detected
                      </td>
                      <td
                        class="text-top"
                      >
                        LAB DEVICE: BIOFIRE速 FILMARRAY速 2.0 SYSTEM
                      </td>
                      <td
                        class="text-top"
                      >
                        <span
                          class="text-italic text-base"
                        >
                          No data
                        </span>
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="text-top"
                      >
                        Plesiomonas shigelloides, NAAT
                      </td>
                      <td
                        class="text-top"
                      >
                        Not Detected
                      </td>
                      <td
                        class="text-top"
                      >
                        Not Detected
                      </td>
                      <td
                        class="text-top"
                      >
                        LAB DEVICE: BIOFIRE速 FILMARRAY速 2.0 SYSTEM
                      </td>
                      <td
                        class="text-top"
                      >
                        <span
                          class="text-italic text-base"
                        >
                          No data
                        </span>
                      </td>
                    </tr>
                  </tbody>
                </table>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Analysis Time
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      09/28/2022 1:59 PM PDT
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Collection Time
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      09/28/2022 8:51 PM UTC
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Received Time
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      09/28/2022 8:51 PM UTC
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Specimen (Source)
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      Stool
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Anatomical Location/Laterality
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      STOOL SPECIMEN / Unknown
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Collection Method/Volume
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      <span
                        class="no-data text-italic text-base"
                      >
                        No data
                      </span>
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Resulting Agency Comment
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      <span
                        class="no-data text-italic text-base"
                      >
                        No data
                      </span>
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Authorizing Provider
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      Ambhp1 Test MD
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Result Type
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      MICROBIOLOGY - GENERAL ORDERABLES
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Narrative
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      <span
                        class="no-data text-italic text-base"
                      >
                        No data
                      </span>
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <h4
            class="usa-summary-box__heading padding-y-105"
            id="lab-results-from-vumc-cerner-lab"
          >
            Lab Results from 
                VUMC CERNER LAB
          </h4>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title"
                >
                  Lab Performing Name
                </div>
                <div
                  class="grid-col-auto maxw7 text-pre-line"
                >
                  VUMC CERNER LAB
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title"
                >
                  Lab Address
                </div>
                <div
                  class="grid-col-auto maxw7 text-pre-line"
                >
                  4605 TVC VUMC
1301 Medical Center Drive
NASHVILLE, TN
37232-5310, USA
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title"
                >
                  Number of Results
                </div>
                <div
                  class="grid-col-auto maxw7 text-pre-line"
                >
                  1
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div
              class="display-flex"
            >
              <div
                class="margin-left-auto padding-top-1"
              >
                <button
                  class="usa-button usa-button--unstyled"
                  data-testid="button"
                  id="lab-info-expand-button"
                  type="button"
                >
                  Expand all labs
                </button>
                <span
                  class="vertical-line"
                />
                <button
                  class="usa-button usa-button--unstyled"
                  data-testid="button"
                  id="lab-info-collapse-button"
                  type="button"
                >
                  Collapse all labs
                </button>
              </div>
            </div>
            <div
              class="usa-accordion accordion-rr accordion_22c6cdd0-bde1-e220-9ba4-2c2802f795ad margin-bottom-3"
              data-testid="accordion"
            >
              <h5
                class="usa-accordion__heading acc_item_22c6cdd0-bde1-e220-9ba4-2c2802f795ad"
              >
                <button
                  aria-controls="Cytogenomic SNP microarray"
                  aria-expanded="false"
                  class="usa-accordion__button"
                  data-testid="accordionButton_Cytogenomic SNP microarray"
                  type="button"
                >
                  Cytogenomic SNP microarray
                </button>
              </h5>
              <div
                class="usa-accordion__content usa-prose acc_item_22c6cdd0-bde1-e220-9ba4-2c2802f795ad"
                data-testid="accordionItem_Cytogenomic SNP microarray"
                hidden=""
                id="Cytogenomic SNP microarray"
              >
                <table
                  class="usa-table usa-table--borderless width-full src-components-Table-Table-module__fixed--Eq00W table-caption-margin margin-y-0"
                  data-testid="table"
                >
                  <thead>
                    <tr>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Component
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Value
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Ref Range
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Test Method
                      </th>
                      <th
                        class="tableHeader"
                        scope="col"
                      >
                        Lab Comment
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td
                        class="text-top"
                      >
                        Observation display
                      </td>
                      <td
                        class="text-top"
                      >
                        <span
                          class="text-italic text-base"
                        >
                          No data
                        </span>
                      </td>
                      <td
                        class="text-top"
                      >
                        <span
                          class="text-italic text-base"
                        >
                          No data
                        </span>
                      </td>
                      <td
                        class="text-top"
                      >
                        <span
                          class="no-data text-italic text-base"
                        >
                          No data
                        </span>
                      </td>
                      <td
                        class="text-top"
                      >
                        <button
                          aria-controls="hidden-comment-4"
                          aria-expanded="false"
                          class="usa-button usa-button--unstyled"
                          data-testid="button"
                          type="button"
                        >
                          View
                           
                          comment
                        </button>
                      </td>
                    </tr>
                    <tr
                      hidden=""
                      id="hidden-comment-4"
                    >
                      <td
                        class="hideableData"
                        colspan="5"
                      >
                        Lab comment
                      </td>
                    </tr>
                  </tbody>
                </table>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Analysis Time
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      <span
                        class="no-data text-italic text-base"
                      >
                        No data
                      </span>
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Collection Time
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      04/04/2022 4:55 PM UTC
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Received Time
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      04/18/2022 7:46 PM UTC
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Specimen (Source)
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      Blood
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Anatomical Location/Laterality
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      <span
                        class="no-data text-italic text-base"
                      >
                        No data
                      </span>
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Collection Method/Volume
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      Collection / Unknown
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Resulting Agency Comment
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      testing comment
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Authorizing Provider
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      Carda Testprovqa MD
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Result Type
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      LAB GENETIC TESTING
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
                <div>
                  <div
                    class="grid-row"
                  >
                    <div
                      class="data-title"
                    >
                      Narrative
                    </div>
                    <div
                      class="grid-col-auto maxw7 text-pre-line lab-text-content"
                    >
                      VUMC CERNER LAB - 04/18/2022 2:57 PM CDT  MICROARRAY REPORT NARRATIVE
                    </div>
                  </div>
                  <div
                    class="section__line_gray"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
